### PR TITLE
Use semeio for parameter parsing, make nesting explicit

### PR DESCRIPTION
**Pull request by berland**
The nesting of parameters dictionary is specific to fmu-dataio. Make it more explicit
by putting it in a different function

### CODE REVIEW

File: tests/test_fmu_dataio_utils.py [hunk 1]
```diff
@@ -128,22 +128,52 @@ def test_uuid_from_string():
...
-    """Testing parsing of justified parameters.txt to JSON"""
+    """Testing parsing of justified parameters.txt into nested dictionary"""
 
     ptext = "tests/data/drogon/ertrun1/realization-0/iter-0/parameters_justified.txt"
 
-    res = _utils.read_parameters_txt(ptext)
+    res = _utils.nested_parameters_dict(_utils.read_parameters_txt(ptext))
 
     assert res["SENSNAME"] == "rms_seed"
     assert res["GLOBVAR"]["VOLON_PERMH_CHANNEL"] == 1100
```
Comment (jcrivenaes) [inline tests/test_fmu_dataio_utils.py:179]: Further up the form is now `res["GLOBVAR:VOLON_PERMH_CHANNEL"]`. Put here is `res["GLOBVAR"]["VOLON_PERMH_CHANNEL"]`?

Comment (berland) [reply tests/test_fmu_dataio_utils.py:179]: Yes, that is what the `nested_parameters_dict()` function does, splitting by colons.

**Review by jcrivenaes — APPROVED** [review]
LGTM

**Comment by berland** [comment]
This PR maybe changes some behaviour of dict ordering, as the previous code used `OrderedDict`. This was apparently not tested. Is that an important feature to be conserved?

**Comment by jcrivenaes** [comment]
> This PR maybe changes some behaviour of dict ordering, as the previous code used `OrderedDict`. This was apparently not tested. Is that an important feature to be conserved?

There may actually be no need for OrderedDict() when >= python 3.6. cf. https://softwaremaniacs.org/blog/2020/02/05/dicts-ordered/

**Comment by perolavsvendsen** [comment]
> > This PR maybe changes some behaviour of dict ordering, as the previous code used `OrderedDict`. This was apparently not tested. Is that an important feature to be conserved?
> 
> There may actually be no need for OrderedDict() when >= python 3.6. cf. https://softwaremaniacs.org/blog/2020/02/05/dicts-ordered/

Regardless, I cannot see that dict ordering should be important to preserve, other than for readability.

**Pull request merged**